src/NavBar.js

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import {
+  AppBar,
+  Button,
+  Toolbar,
+  Typography,
+} from '@material-ui/core';
+import { useAuth } from './AuthProvider';
+
+function NavBar() {
+  const auth = useAuth();
+  const handleSignOut = () => {
+    auth.signOut();
+  };
```

Comment:
You can make this shorter like this if you want: `const handleSignOut = () => auth.signOut();`